File: java/src/main/java/sn/brasilburger/entity/Complement.java
package sn.brasilburger.entity;

import sn.brasilburger.entity.enums.TypeComplement;

public class Complement {

    private int id;
    private String nom;
    private double prix;
    private TypeComplement typeComplement;
    private String image;
    private boolean actif;

    public Complement() {
        this.actif = true;
    }

    public Complement(String nom, double prix, TypeComplement typeComplement, String image) {
        this.nom = nom;
        this.prix = prix;
        this.typeComplement = typeComplement;
        this.image = image;
        this.actif = true;
    }

    // GETTERS / SETTERS

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }

    public double getPrix() { return prix; }
    public void setPrix(double prix) { this.prix = prix; }

    public TypeComplement getTypeComplement() { return typeComplement; }
    public void setTypeComplement(TypeComplement typeComplement) { this.typeComplement = typeComplement; }

    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    public boolean isActif() { return actif; }
    public void setActif(boolean actif) { this.actif = actif; }

    @Override
    public String toString() {
        return "Complement{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", prix=" + prix +
                ", type=" + typeComplement +
                ", actif=" + actif +
                '}';
    }
}

File: symfony/brasil-burger-symfony/src/Entity/Complement.php
<?php

namespace App\Entity;

use App\Repository\ComplementRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ComplementRepository::class)]
#[ORM\Table(name: 'complements')]
class Complement
{
    public const TYPE_BOISSON = 'BOISSON';
    public const TYPE_FRITES = 'FRITES';

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(name: 'nom', length: 150)]
    private ?string $name = null;


    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(name: 'type_complement', length: 20)]
    private ?string $typeComplement = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(options: ['default' => true])]
    private ?bool $actif = true;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getTypeComplement(): ?string
    {
        return $this->typeComplement;
    }

    public function setTypeComplement(string $typeComplement): static
    {
        $this->typeComplement = $typeComplement;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): static
    {
        $this->image = $image;

        return $this;
    }

    public function isActif(): ?bool
    {
        return $this->actif;
    }

    public function setActif(?bool $actif): static
    {
        $this->actif = $actif;

        return $this;
    }
    
}
